### ISSUE TITLE
Proposal: Runtime upgrade

### ISSUE THREAD

**Issue by jodator** [Joystream/pioneer#802]
@[figma](https://www.figma.com/file/pgNl83wgAub186krVkdiuo/Proposals?node-id=267%3A3163)

Handle proposals specific properties:

- [ ] File preview / download

Notes:
- See https://github.com/Joystream/pioneer/issues/872 as well

**Comment by jodator** [comment]
Upss... wrong issue in the PR comment

**Comment by p-sad** [comment]
A couple of observations:

I used an `ArrayBuffer` for the object seeded to and fetched from the MirageJS server, and while Mirage accepts it without a hitch, the final implementation on Hydra might be different. I looked at the Hydra repo and it doesn't seem like the `Bytes` type is well defined yet.

One way that would allow bytecode to be downloaded would be by using a [data URL](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URIs) that contains a Base64 string representation of the bytecode.
There also exist packages that help create an in-memory filesystem, but I'm not sure whether they would be useful in this particular case (or if they would still require conversion to a data URL).
